### ISSUE TITLE
Data race in seriesCursor.Next

### ISSUE THREAD

**Issue by jwilder** [influxdb/influxdb#3667]
To reproduce:
Enable TestServer_ContinuousQuery_Deadlock

go test -run TestServer_ContinuousQuery_Deadlock -v -race ./cmd/influxd/run/

```
WARNING: DATA RACE
Read by goroutine 49:
  encoding/binary.bigEndian.Uint64()
      /usr/local/go/src/encoding/binary/binary.go:110 +0x46
  github.com/influxdb/influxdb/tsdb.btou64()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:835 +0x6d
  github.com/influxdb/influxdb/tsdb.(*seriesCursor).Next()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:760 +0xf4
  github.com/influxdb/influxdb/tsdb.(*tagSetCursor).Next()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:669 +0x202
  github.com/influxdb/influxdb/tsdb.func·006()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:360 +0x2a6
  github.com/influxdb/influxdb/tsdb.(*aggTagSetCursor).Next()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:561 +0x55
  github.com/influxdb/influxdb/influxql.MapSum()
      /Users/jason/go/src/github.com/influxdb/influxdb/influxql/functions.go:432 +0x72
  github.com/influxdb/influxdb/tsdb.(*LocalMapper).nextChunkAgg()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:370 +0x1202
  github.com/influxdb/influxdb/tsdb.(*LocalMapper).NextChunk()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/mapper.go:253 +0xc4
  github.com/influxdb/influxdb/tsdb.(*StatefulMapper).NextChunk()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/executor.go:43 +0x77
  github.com/influxdb/influxdb/tsdb.(*Executor).executeAggregate()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/executor.go:376 +0xaea

Previous write by goroutine 47:
  [failed to restore the stack]

Goroutine 49 (running) created at:
  github.com/influxdb/influxdb/tsdb.(*Executor).Execute()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/executor.go:90 +0x147
  github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/query_executor.go:261 +0xba
  github.com/influxdb/influxdb/tsdb.func·008()
      /Users/jason/go/src/github.com/influxdb/influxdb/tsdb/query_executor.go:146 +0x97a

Goroutine 47 (finished) created at:
  github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard()
      /Users/jason/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:271 +0x42c
  github.com/influxdb/influxdb/cluster.func·001()
      /Users/jason/go/src/github.com/influxdb/influxdb/cluster/points_writer.go:205 +0x149
```

**Comment by otoolep** [comment]
Fixed @DanielMorsing ?

**Comment by DanielMorsing** [comment]
closed by #3671
